neo/io/kwikio.py

```diff
@@ -0,0 +1,207 @@
+
+    def read_block(self,
+                     lazy=False,
+                     cascade=True,
+                     dataset=0,
+                     channel_index=None,
+                     sampling_rate=None
+                    ):
+        """
+        Arguments:
+            Channel_index: can be int, iterable or None to select one, many or all channel(s)
+            dataset: points to a specific dataset in the .kwik and .raw.kwd file,
+                     however this will be an issue to change in e.g. OpenElectrophy or Spykeviewer
+            sampling_rate: None or float/int if not None raw data will be downsampled
+        """
+
+        attrs = {}
+        attrs['kwik'] = self._kwik['recordings'][str(dataset)].attrs
+        attrs['kwd'] = self._kwd['recordings'][str(dataset)].attrs
+        attrs['shape'] = self._kwd['recordings'][str(dataset)]['data'].shape
+        try:
+            attrs['app_data'] = self._kwd['recordings'][str(dataset)]['application_data'].attrs # TODO: find bitvolt conversion in phy generated data
+        except:
+            attrs['app_data'] = False
+
+        blk = Block()
+        if cascade:
+            seg = Segment( file_origin=self._filename ,name='test' )
+            blk.segments += [ seg ]
+            if channel_index:
+                if type(channel_index) is int: channel_index = [channel_index]
+            else:
+                channel_index = range(0,attrs['shape'][1])
+
+            for idx in channel_index:
+                # read nested analosignal
+                ana = self._read_traces(attrs=attrs,
+                                        channel_index=idx,
+                                        lazy=lazy,
+                                        cascade=cascade,
+                                        dataset=dataset,
+                                        sampling_rate=sampling_rate,
+                                         )
+                seg.analogsignals += [ ana ]
+
+            seg.duration = (attrs['shape'][0]
+                          / attrs['kwik']['sample_rate']) * pq.s
+
+            neo.tools.populate_RecordingChannel(blk)
+        blk.create_many_to_one_relationship()
+        return blk
+
+    def _read_traces(self,
+                      attrs,
+                      channel_index,
+                      lazy=False,
+                      cascade=True,
+                      dataset=0,
+                      sampling_rate=None,
+                      ):
+        """
+        read raw traces with given sampling_rate, if sampling_rate is None
+        default from acquisition system is given. channel_index can be int or
+        iterable, if None all channels are read
+        """
+
+        if sampling_rate:
+            sliceskip = int(attrs['kwik']['sample_rate']/sampling_rate)
+        else:
+            sliceskip = 1
+            sampling_rate = attrs['kwik']['sample_rate']
+
```

Comment:
I do not like very this downsampling inside the IOs.
It should be outside the IOs.
Furthermore, this is NOT a downsampling, it is a decimatation but when you decimate you MUST filter you signal before this decimation otherwise you make an aliasing mistake (a big one).

See scipy.signal.decimate https://github.com/scipy/scipy/blob/master/scipy/signal/signaltools.py.

So I suggest 1) not put this inside the IO 2) never do again a decimation without filtering before!!!!!